ethcore/res/ethereum/tests-issues/currents.json

```diff
@@ -0,0 +1,190 @@
+                                        "chain": "Constantinople (test)"
+                                },
+                                "LoopCallsDepthThenRevert2": {
+                                        "subnumbers": ["1"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "RevertDepth2": {
+                                        "subnumbers": ["1"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "RevertRemoteSubCallStorageOOG2": {
+                                        "subnumbers": ["1","2"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "RevertDepthCreateAddressCollision": {
+                                        "subnumbers": ["3","4"],
+                                        "chain": "Constantinople (test)"
+                                }
+                        }
+                },
+                {
+                        "reference": "9568",
+                        "failing": "stStaticCall",
+                        "subtests": {
+                                "static_RevertDepth2": {
+                                        "subnumbers": ["1","3"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "static_CheckOpcodes4": {
+                                        "subnumbers": ["3"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "static_CheckOpcodes3": {
+                                        "subnumbers": ["5","6","7","8"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "static_callBasic": {
+                                        "subnumbers": ["1","2"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "static_CheckOpcodes2": {
+                                        "subnumbers": ["5","6","7","8"],
+                                        "chain": "Constantinople (test)"
+                                },
+                                "static_callCreate": {
+                                        "subnumbers": ["2"],
+                                        "chain": "Constantinople (test)"
+                                }
+ 
+                        }
+                },
+                {
+                        "reference": "https://github.com/ethereum/tests/issues/512",
+                        "failing": "stZeroKnowledge",
+                        "subtests": {
+                                "pointAddTrunc": {
```

Comment:
I think those one are all actually passing except `pointAdd`? @cheme Do you mind to check this?

Reply:
Not passing :( , there is something bad with our current code : if you run the test without skipping anything it fails on only one test, but as soon as you skip it, it reveals others failure.

Reply:
Got it! Can you also cherry-pick https://github.com/paritytech/parity-ethereum/pull/9577/commits/16f661a59518bf0d949b0b15ea0191fa4df5ba95? Our previous definition of `modexp` and `bn128_mul` has wrong gas price. I think it may fix some other tests.

Reply:
Indeed, it does reduce the number of failing zk tests. Many still don't pass, but it is due to https://github.com/ethereum/tests/issues/512 (with 1283 disabled everything is fine).